cg/meta/backup/backup.py

```diff
@@ -62,33 +77,211 @@ def fetch_flow_cell(
+            if error.returncode == RETURN_WARNING:
+                LOG.warning(
+                    "WARNING for retrieval of encryption key of flow cell %s, please check "
+                    "dsmerror.log",
+                    flow_cell_obj.name,
+                )
+            else:
+                LOG.error("%s: key retrieval failed", flow_cell_obj.name)
+                if not self.dry_run:
+                    flow_cell_obj.status = FlowCellStatus.REQUESTED
+                    self.status.commit()
+                raise error
 
+    def retrieve_archived_flow_cell(
+        self, archived_flow_cell: Path, flow_cell_obj: models.Flowcell, run_dir: Path
+    ):
+        """Attempt to retrieve an archived flow cell"""
         try:
-            self.pdc.retrieve_flow_cell(
-                flow_cell_id=flow_cell_obj.name,
-                sequencer_type=flow_cell_obj.sequencer_type,
-                root_dir=self.root_dir,
-                dry=dry_run,
+            self.retrieve_archived_file(
+                archived_file=archived_flow_cell,
+                run_dir=run_dir,
             )
-            if not dry_run:
+            if not self.dry_run:
                 flow_cell_obj.status = FlowCellStatus.RETRIEVED
                 self.status.commit()
                 LOG.info(
                     "Status for flow cell %s set to %s", flow_cell_obj.name, flow_cell_obj.status
                 )
         except subprocess.CalledProcessError as error:
-            LOG.error("%s: retrieval failed", flow_cell_obj.name)
-            if not dry_run:
-                flow_cell_obj.status = FlowCellStatus.REQUESTED
-                self.status.commit()
-            raise error
+            if error.returncode == RETURN_WARNING:
+                LOG.warning(
+                    "WARNING for retrieval of flow cell %s, please check dsmerror.log",
+                    flow_cell_obj.name,
+                )
+                if not self.dry_run:
+                    flow_cell_obj.status = FlowCellStatus.RETRIEVED
+                    self.status.commit()
+                    LOG.info(
+                        "Status for flow cell %s set to %s",
+                        flow_cell_obj.name,
+                        flow_cell_obj.status,
+                    )
+            else:
+                LOG.error("%s: run directory retrieval failed", flow_cell_obj.name)
+                if not self.dry_run:
+                    flow_cell_obj.status = FlowCellStatus.REQUESTED
+                    self.status.commit()
+                raise error
 
-        return get_elapsed_time(start_time=start_time)
+    def query_pdc_for_flow_cell(self, flow_cell_id) -> list:
+        """Query PDC for a given flow cell id"""
+        search_pattern = EncryptionDirsAndFiles.ENCRYPT_DIR + ASTERISK + flow_cell_id + ASTERISK
+        self.pdc.query_pdc(search_pattern=search_pattern)
+        query: list = self.pdc.process.stdout.split(NEW_LINE)
+        return query
```

Comment:
This query could be ordered by date and return only the latest entries?

Reply:
Created issue with this comment: https://github.com/Clinical-Genomics/cg/issues/1515